include/rapidjson/rapidjson.h

```diff
@@ -617,6 +617,22 @@ RAPIDJSON_NAMESPACE_END
 #define RAPIDJSON_NOEXCEPT_ASSERT(x) RAPIDJSON_ASSERT(x)
 #endif // RAPIDJSON_ASSERT_THROWS
 
+///////////////////////////////////////////////////////////////////////////////
+// malloc/realloc/free
+
+#ifndef RAPIDJSON_MALLOC
+///! customization point for global \c malloc
+#define RAPIDJSON_MALLOC(size) std::malloc(size)
+#endif
+#ifndef RAPIDJSON_REALLOC
+///! customization point for global \c realloc
+#define RAPIDJSON_REALLOC(ptr, new_size) std::realloc(ptr, new_size)
+#endif
+#ifndef RAPIDJSON_FREE
+///! customization point for global \c free
+#define RAPIDJSON_FREE(ptr) std::free(ptr)
+#endif
```

Comment:
Better perhaps would be to expose some externally accessible hooks that could be set in code by the user of rapidjson library. By default they could point to `std::` functions and, if set by user, could point to any user-provided function.

Reply:
Using function pointers for this seems adding unnecessary overheads.
I think macro customization is sufficient for most users. And they can define a macro to call a function that can be further redirect via function pointer.

Reply:
I agree, I did not see much benefit in adding function pointers. If desired, library owners can chose to define the macro such that it points to a customizable allocator.